src/test/java/com/humio/connect/hec/EndToEndJsonTest.java

```diff
@@ -287,24 +286,35 @@ public static void teardown() {
     }
 
     private static String getIngestToken() throws IOException {
-        System.out.println("extracting " + HUMIO_INDEX + " ingest key from global data snapshot...");
-        String s = new String(Files.readAllBytes(Paths.get("humio-data/global-data-snapshot.json")));
-        JsonObject data = new JsonParser().parse(s).getAsJsonObject();
-        JsonObject dataspaces = data.getAsJsonObject("dataspaces");
-        Set<Map.Entry<String, JsonElement>> entries = dataspaces.entrySet();
-        for(Map.Entry<String, JsonElement> entry : entries) {
-            String key = entry.getKey();
-            JsonObject obj = entry.getValue().getAsJsonObject();
-            if (key.startsWith(HUMIO_INDEX) && key.indexOf("LocalHostRoot") == -1) {
-                System.out.println("located " + HUMIO_INDEX + " key: " + key);
-                JsonObject sandbox = dataspaces.getAsJsonObject(key);
-                JsonObject entity = sandbox.getAsJsonObject("entity");
-                JsonArray tokens = entity.getAsJsonArray("ingestTokens");
-                JsonObject token = tokens.get(0).getAsJsonObject();
-                return token.get("token").getAsString();
+        System.out.println("getting " + HUMIO_INDEX + " ingest token via graphql...");
+
+        RequestBody body = RequestBody.create(
+                MediaType.parse("application/json"),
+                "{ \"query\": \"query { repositories { name ingestTokens { token } } }\"} "
+        );
+
+        Request request = new Request.Builder()
+                .url("http://localhost:"+HUMIO_PORT+"/graphql")
+                .post(body)
+                .build();
+
+        Response response = new OkHttpClient().newCall(request).execute();
```

Comment:
Response is Closeable, so I think this should be wrapped in a try-with-resources.